### PR TITLE
media-gfx/gif2png: bump version to 3.0.0

### PR DESCRIPTION
**Pull request by kernlpanic**
Closes: https://bugs.gentoo.org/724518
Package-Manager: Portage-3.0.28, Repoman-3.0.3
Signed-off-by: Dennis Eisele <mail@dennis-eisele.de>

### CODE REVIEW

File: media-gfx/gif2png/gif2png-3.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	>=dev-lang/go-1.17.5"
+
+# Tests are known to fail, #688702
+# EOL; Upstream has rewritten gif2png in Go
+RESTRICT=test
+
+src_prepare() {
+	default
+	cd "${S}"
+	go mod init gif2png
```
Comment (thesamesam) [inline media-gfx/gif2png/gif2png-3.0.0.ebuild:28]: Always need a `|| die` on (external) commands which may fail, `cd` counts here too.

File: media-gfx/gif2png/gif2png-3.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+SLOT="0"
+KEYWORDS="~amd64 -ppc ~ppc64 -sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+
+RDEPEND=">=media-libs/libpng-1.2:0=
+	sys-libs/zlib:="
+DEPEND="${RDEPEND}
+	>=dev-lang/go-1.17.5"
+
+# Tests are known to fail, #688702
+# EOL; Upstream has rewritten gif2png in Go
```
Comment (thesamesam) [inline media-gfx/gif2png/gif2png-3.0.0.ebuild:22]: This comment at least isn't relevant anymore. Do tests pass now?

File: media-gfx/gif2png/gif2png-3.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+SRC_URI="http://catb.org/~esr/${PN}/${P}.tar.gz"
+
+LICENSE="ZLIB"
+SLOT="0"
+KEYWORDS="~amd64 -ppc ~ppc64 -sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+
+RDEPEND=">=media-libs/libpng-1.2:0=
+	sys-libs/zlib:="
+DEPEND="${RDEPEND}
+	>=dev-lang/go-1.17.5"
```
Comment (thesamesam) [inline media-gfx/gif2png/gif2png-3.0.0.ebuild:19]: Put go in BDEPEND.

**Review by thesamesam — CHANGES_REQUESTED** [review]

**Comment by kernlpanic** [comment]
@thesamesam I updated the ebuild accordingly

**Comment by thesamesam** [comment]
Do you not have ```FEATURES="network-sandbox"``` enabled?

```
>>> Unpacking gif2png-3.0.0.tar.gz to /var/tmp/portage/media-gfx/gif2png-3.0.0/work
>>> Source unpacked in /var/tmp/portage/media-gfx/gif2png-3.0.0/work
/var/tmp/portage/media-gfx/gif2png-3.0.0/work/gif2png-3.0.0 /var/tmp/portage/media-gfx/gif2png-3.0.0/work
/var/tmp/portage/media-gfx/gif2png-3.0.0/work
>>> Preparing source in /var/tmp/portage/media-gfx/gif2png-3.0.0/work/gif2png-3.0.0 ...
go: creating new go.mod: module gif2png
go: to add module requirements and sums:
	go mod tidy
go get: module golang.org/x/crypto/ssh/terminal: Get "https://proxy.golang.org/golang.org/x/crypto/ssh/terminal/@v/list": dial tcp [2a00:1450:400b:c01::8d]:443: connect: network is unreachable
 * ERROR: media-gfx/gif2png-3.0.0::gentoo failed (prepare phase):
 *   failed to get terminal package
 *
 * Call stack:
 *     ebuild.sh, line 127:  Called src_prepare
 *   environment, line 1256:  Called die
 * The specific snippet of code:
 *       go get golang.org/x/crypto/ssh/terminal || die "failed to get terminal package";
```

**Comment by juippis** [comment]
So that ^ error looks like upstream doesn't package required go modules, but instead depend on getting them via network while building. Please see other go-module ebuilds how it's handled, also, you can package / ask upstream to package the modules. Some upstreams do that.

Although I haven't checked the upstream build system for this one, all the above is just guesswork. 

Let us know if you need more help and we'll dig into it!

**Comment by thesamesam** [comment]
Let's close for now (as per discussion on IRC) pending resolution of the upstream bug: https://gitlab.com/esr/gif2png/-/issues/14

**Pull request closed without merging**